plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableHandle.java

```diff
@@ -42,7 +42,8 @@
     private final TableType tableType;
     private final Optional<Long> snapshotId;
     private final String tableSchemaJson;
-    private final String partitionSpecJson;
+    // The partition spec information is needed only when doing DML operations
+    private final Optional<String> partitionSpecJson;
```

Comment:
We should rather explain why it's Optional, ie. what's the meaning of empty.

```
// Empty means the partitioning spec is not known (can be the case for certain time travel queries).
```

Current usages are easy to find out, and can change, so we don't need to document them here.